secio/src/codec/hmac_compat/mod.rs

```diff
@@ -17,7 +21,7 @@ mod test {
     use rand::Rng;
 
     #[test]
-    fn test_mac() {
+    fn test_mac_oneshot() {
```

Comment:
Recommend using proptest here, which can save the test case in case the test fails.